onebusaway-android/src/main/java/org/onebusaway/android/util/LocationUtils.java

```diff
@@ -214,6 +219,37 @@ public static GoogleApiClient getGoogleApiClientWithCallbacks(Context context) {
     }
 
     /**
+     * Checks if string is valid zip code.
+     */
+    public static boolean isValidZipCode(String zip)
+    {
+        Pattern pattern = Pattern.compile("^\\d{5}(?:-\\d{4})?$");
+        return pattern.matcher(zip).matches();
+    }
+
+    /**
```

Comment:
For this Javadoc, please add that this method makes a call to a server (in platform in `geocoder.getFromLocationName()`), and could be a long running operation, and therefore shouldn't be done on the UI thread.